README.md

```diff
@@ -378,6 +378,25 @@ The default is defined as below.
 #define PCC_FREE(auxil, ptr) free(ptr)
 ```
 
+**`PCC_DEBUG(`**_event_**`,`**_rule_**`,`**_level_**`,`**_pos_**`,`**_buffer_**`,`**_length_**`)`**
+
+The function macro for debugging. Sometimes, especially for complex parsers, it is useful to see how exactly the parser processes the input. This macro is called on important *events* and allows to log or display current state of the parsers. The argument `rule` is a string that contains name of currently evaluated rule. Non-negative integer `level` is a specifying how deep in the rule hierarchy the parser currently is. Argument `pos` holds position from the start of current context in bytes. In case of `event == DBG_MATCH` argument `buffer` holds matched input and `length` is its size. for other events `buffer` and `length` describe part of currently loaded input, that is used to evaluate rule.
+
+There are currently three supported events:
+ - `DBG_EVALUATE` - called when the parser starts to evaluate `rule`
+ - `DBG_MATCH` - called when `rule` is matched, at which point buffer holds entire matched string
+ - `DBG_NOMATCH` - called when the parsers determines that the input does not match currently evaluated `rule`
+
+Very simple implementation could look like this:
+
+```C
+static const char* dbg_str[] = {"Evaluating rule", "Matched rule", "Abandoning rule"};
+#define PCC_DEBUG(event, rule, level, pos, length, buffer) \
+    fprintf(stderr, "%*s%s %s @%d [%.*s]\n", level * 2, "", dbg_str[event], rule, pos, length, buffer);
```

Comment:
This example assumes that `DBG_*` macro values are numbered as 0, 1, and 2.
I think it would be better not to let uses have this assumption.

Reply:
The trailing `;` should be removed for a function macro.

Reply:
I have changed it from array to a function with switch statement. But it seems kind of long and "inelegant", compared to previous version. What if we made it part of the interface and documented it in the README, that the values are continuous integers? I think many users would figure out and use the same simplification as me anyway :slightly_smiling_face:

Reply:
OK, I agree if the macro values are explicitly shown to users in the README.txt.

Reply:
I'll pull this PR first, and modify it back to adopt your proposal.

Reply:
You are way too fast for me :laughing: Thank you for finishing this. And also thank you for fixing my english in the README.